### PR TITLE
feat(web-components): add appearance prop to tab panel

### PR DESCRIPTION
**Pull request by GCHQ-Developer-530**
## Summary of the changes
Add appearance prop to tab panel and update tab context to change the appearance of tab panel

## Related issue
#112

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [x] I have manually accessibility tested any changes, if relevant.
- [x] I have ensured any changes match the Figma component library.

### CODE REVIEW

**Comment by GCHQ-Developer-112** [comment]
Sorry, I forgot to add this to the ticket but can you flip the appearance prop on this component. For this component, appearance: dark means the background is dark but for other components appearance="dark" means the component's text is dark. In line with [guidance on tabs](https://design.sis.gov.uk/components/tabs),  the behaviour of tabs should be the latter

**Comment by GCHQ-Developer-530** [comment]
> Sorry, I forgot to add this to the ticket but can you flip the appearance prop on this component. For this component, `appearance="dark"` means the background is dark but for other components, `appearance="dark"` means the component's text is dark. In line with [guidance on tabs](https://design.sis.gov.uk/components/tabs), the behaviour of tabs should be the latter

This would mean that it's out of line with the tab component itself, should I update the tabs component as well?

**Comment by GCHQ-Developer-112** [comment]
> > Sorry, I forgot to add this to the ticket but can you flip the appearance prop on this component. For this component, `appearance="dark"` means the background is dark but for other components, `appearance="dark"` means the component's text is dark. In line with [guidance on tabs](https://design.sis.gov.uk/components/tabs), the behaviour of tabs should be the latter
> 
> This would mean that it's out of line with the tab component itself, should I update the tabs component as well?

Yes I think that would be best. I'm going to double check that changing the appearance doesn't constitute a breaking change

**Comment by GCHQ-Developer-112** [comment]
> > > Sorry, I forgot to add this to the ticket but can you flip the appearance prop on this component. For this component, `appearance="dark"` means the background is dark but for other components, `appearance="dark"` means the component's text is dark. In line with [guidance on tabs](https://design.sis.gov.uk/components/tabs), the behaviour of tabs should be the latter
> > 
> > 
> > This would mean that it's out of line with the tab component itself, should I update the tabs component as well?
> 
> Yes I think that would be best. I'm going to double check that changing the appearance doesn't constitute a breaking change

UPDATE: this doesn't constitute a breaking change so we're good to implement it. Are you happy to make the changes?

**Comment by GCHQ-Developer-530** [comment]
> > > > Sorry, I forgot to add this to the ticket but can you flip the appearance prop on this component. For this component, `appearance="dark"` means the background is dark but for other components, `appearance="dark"` means the component's text is dark. In line with [guidance on tabs](https://design.sis.gov.uk/components/tabs), the behaviour of tabs should be the latter
> > > 
> > > 
> > > This would mean that it's out of line with the tab component itself, should I update the tabs component as well?
> > 
> > 
> > Yes I think that would be best. I'm going to double check that changing the appearance doesn't constitute a breaking change
> 
> UPDATE: this doesn't constitute a breaking change so we're good to implement it. Are you happy to make the changes?

Yep I made the changes in my last commit :) they're all ready for reviewing now

**Review by GCHQ-Developer-112 — COMMENTED** [review]
There are some cases where you just want the tab group to be dark. Can we add some functionality for this? In the cases where the appearance prop is added to `ic-tab-context`, we may want this to override the appearance prop on the tab group/tab panel. Not sure if this functionality would be best on the `ic-tab-context` or on the `ic-tab group / ic-tab panel`

Screenshot 2022-11-30 at 09 31 36

**Comment by GCHQ-Developer-530** [comment]
> There are some cases where you just want the tab group to be dark. Can we add some functionality for this? In the cases where the appearance prop is added to `ic-tab-context`, we may want this to override the appearance prop on the tab group/tab panel. Not sure if this functionality would be best on the `ic-tab-context` or on the `ic-tab group / ic-tab panel`
> 
> Screenshot 2022-11-30 at 09 31 36

This functionality is already possible, the user would just set the tab group appearance to be dark and not set the entire tab context appearance as dark. The tab context also already overrides the tab group/tab panel appearance so if tab context is dark, even if the user sets tab panel as light, it will still show as dark :)

**Comment by GCHQ-Developer-112** [comment]
> > There are some cases where you just want the tab group to be dark. Can we add some functionality for this? In the cases where the appearance prop is added to `ic-tab-context`, we may want this to override the appearance prop on the tab group/tab panel. Not sure if this functionality would be best on the `ic-tab-context` or on the `ic-tab group / ic-tab panel`
> > Screenshot 2022-11-30 at 09 31 36
> 
> This functionality is already possible, the user would just set the tab group appearance to be dark and not set the entire tab context appearance as dark. The tab context also already overrides the tab group/tab panel appearance so if tab context is dark, even if the user sets tab panel as light, it will still show as dark :)

Oh ok, maybe I'm looking at the wrong branch 🫣. Could you update the React story as well please?

File: packages/react/src/stories/ic-tabs.stories.mdx [hunk 1]
```diff
@@ -204,7 +204,7 @@ export const ControlledTabsExample = () => {
 
 <Canvas>
   <Story name="Dark" parameters={{ backgrounds: { default: "dark" } }}>
```
Comment (GCHQ-Developer-112) [inline packages/react/src/stories/ic-tabs.stories.mdx:206]:
```suggestion
  <Story name="Light" parameters={{ backgrounds: { default: "dark" } }}>
```

**Comment by GCHQ-Developer-112** [comment]
Last thing, sorry, can the visual tests be updated please? And the old ones with dark in the title be deleted?

**Comment by GCHQ-Developer-530** [comment]
This work has been approved by designers

**Review by GCHQ-Developer-112 — APPROVED** [review]

**Review by jd239 — APPROVED** [review]

**Pull request merged**